### ISSUE TITLE
Document pinned flutter packages

### ISSUE THREAD

**Issue by natebosch** [flutter/website#6162]
### What information is missing?

Some packages dependencies are pinned by `flutter`, `flutter_test`, or other dependencies in the SDK. End users are not able to solve for package versions other than the pinned version which may prevent them from getting new features or bug fixes and cause confusion.

For example https://github.com/dart-lang/async/issues/187

https://github.com/flutter/website/blob/master/src/docs/development/packages-and-plugins/using-packages.md

### How would you like us to fix this problem?

List the packages that are pinned by flutter and describe that the only way to get other versions is to change the version of the flutter SDK.

**Comment by atsansone** [comment]
This was fixed in https://github.com/flutter/website/pull/687. It links to the Dart section on these dependencies. Closing this issue.

**Comment by parlough** [comment]
I think this is discussing a different issue in which the Flutter SDK pins some specific dependencies that users can't override. To solve this they must change their Flutter version. I do think we should still better document this on docs.flutter.dev. 

We recently added https://dart.dev/go/sdk-version-pinning which covers this situation, but some discoverable docs directly on docs.flutter.dev would be helpful.

**Comment by natebosch** [comment]
Yes, some authoritative flutter documentation covering  https://dart.dev/go/sdk-version-pinning is what I was hoping for. #687 is about user pinned packages, and I think we need good documentation for the flutter SDK pinned packages.
